IQSS/dataverse-frontend
# 405 Infinite Scrolling on Dataset Files

**What this PR does / why we need it**:
Adds infinite scrolling to the files list in the files tab from a dataset.

**Which issue(s) this PR closes**:

Closes #405 

**Special notes for your reviewer**:
I have created a configuration file to choose between normal pagination with buttons or infinite scroll.
Many components were reused and adapted with some new prop, others I preferred to make an alternative version specifically for this feature.
Many changes were necessary for the file selection but everything still maintains the same flows.
I also took the opportunity to improve the styles of the criteria form, some elements were collapsing or overflowing in mobile.

**Suggestions on how to test this**:
**Step 1: Run the Development Environment**
1. Execute `npm i`.
2. Navigate with `cd packages/design-system && npm run build`.
3. Return with `cd ../../`.
4. Ensure you have a `.env` file similar to `.env.example`, with the variable `VITE_DATAVERSE_BACKEND_URL=http://localhost:8000`.
5. Navigate with `cd dev-env`.
6. Start the environment using `./run-env.sh unstable`.
7. To verify the environment, visit <http://localhost:8000> and check your local Dataverse installation.

**Step 2: Test the Create Dataset Form**
- Upload multiple files to a dataset in the JSF UI version.
- Scroll to the bottom of the table to load more files.
- Scroll to the bottom and then change any filter, you will be automatically scrolled to the top.
- Load first 10 files, select 10 files and then click the Select All button, after that scroll to the bottom to load more files, these files should be checked as you have selected them before loading them.

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:
Yes.

<img width="1410" alt="Screenshot 2024-06-06 at 09 53 51" src="https://github.com/IQSS/dataverse-frontend/assets/64560524/9959a757-5915-40a3-a5df-bae767d24f93">

**How sticky positioning of all elements works with the infinite scrolling 👇🏼**
<img width="1161" alt="Screenshot 2024-06-06 at 09 58 17" src="https://github.com/IQSS/dataverse-frontend/assets/64560524/393487cc-dbaa-4813-a212-abc348f48f4d">

**Criteria Form layout fixes & table header action buttons only icons for mobile 👇🏼 :**
_Old:_
<img width="437" alt="Screenshot 2024-06-05 at 15 17 48" src="https://github.com/IQSS/dataverse-frontend/assets/64560524/e3d26c9b-9fd7-43f2-92be-f66e6a4d85d2">

_New:_
<img width="439" alt="Screenshot 2024-06-05 at 15 17 59" src="https://github.com/IQSS/dataverse-frontend/assets/64560524/3c6e2eef-ce17-4d31-bb8a-c19ecb7349a8">


**Is there a release notes update needed for this change?**:
No.

**Additional documentation**:
No.
